python/ql/src/Security/CWE-798/HardcodedCredentials.ql

```diff
@@ -30,15 +32,13 @@ predicate fewer_characters_than(StrConst str, string char, float fraction) {
 }
 
 predicate possible_reflective_name(string name) {
-  exists(any(ModuleValue m).attr(name))
+  any(Function f).getName() = name
   or
-  exists(any(ClassValue c).lookup(name))
```

Comment:
I feel like this one got left out in the translation. Do we not have a way of approximating "name of field of some class"? (I would have thought that at least the call-graph calculation would need to know this.)

Reply:
I wrote this in the commit message: `ModuleValue.attr` and `ClassValue.lookup` are approximated by `Function.getName`.

My thinking is that `Function` also covers methods and so `Function.getName` would find the names of all class methods which should be a good approximation of what  `ClassValue.lookup` returns.

Reply:
I see that you mention fields. The [qldoc for `ClasValue.lookup`](https://github.com/github/codeql/blob/main/python/ql/lib/semmle/python/objects/ObjectAPI.qll#L479-L491) only talks about methods, so that is what I tried to approximate. I guess `__dict__` also contains fields, though, so we could try to capture those also. In fact, they might be where one could expect to find hardcoded credentials...

Reply:
Yeah, generally I would expect `lookup` to work for _any_ attribute, not just those that correspond to methods. 

But again, I think it's fine to merge this as-is, and revisit later on.

Reply:
Alright, let's :shipit: